### PR TITLE
fix(injector): source import shadowed results in build error "X is not a type"

### LINKED ISSUE #448: [BUG] SQL registration wrapper can shadow package name
**Version of orchestrion**
v1.0.1

**Describe what happened:**
The orchestrion build exits with an error: 
```bash
orchestrion go build -work -o api cmd/api/main.go
WORK=/var/folders/42/r9szg2bn5xv_z3mrfq0sn2mr0000gn/T/go-build1177213176
# adtech/pkg/core/db
pkg/core/db/metrics.go:32: driver.Driver is not a type
exit status 2
```

**Describe what you expected:**
The generated function wrapper parameters should not shadow names of imported packages. 

**Steps to reproduce the issue:**

Upon close inspection of the work tree, orchestrion generated a function wrapper that shadows the "sql/driver" package. Here is the relevant source code:

```go
package db

import (
	"adtech/pkg/core/mon"
	"database/sql"
	"database/sql/driver"
	"github.com/jmoiron/sqlx"
	"github.com/twinj/uuid"
	"time"
//line <generated>
	__orchestrion_sqltrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql"
)


type metricDriver struct {
	driver.Driver

	metricWriter mon.MetricWriter
}

func newMetricDriver(driver driver.Driver) string {
	mw := mon.NewMetricDaemonWriter()

	id := uuid.NewV4().String()
	md := &metricDriver{
		Driver:       driver,
		metricWriter: mw,
	}

//line <generated>
	func(driverName string, driver driver.Driver) {
		sql.Register(driverName, driver)
		__orchestrion_sqltrace.Register(driverName, driver)
	}(
//line /Users/lukas/Projects/adtech_central/pkg/core/db/metrics.go:31
		id, md)

	return id
}
```

**Additional environment details (Version of Go, Operating System, etc.):**

Version of Go: 1.23.4
OS: macOS Sequoia 15.1.1

### LINKED ISSUE #458: [BUG]  Unable to build against aws-xray-sdk-go package
When attempting to build using orchestrion my package will not build due to this error: 

`github.com/aws/aws-xray-sdk-go@v1.8.5/xray/sql_context.go:62: driver.Driver is not a type`

This does not occur when building without the Orchestrion toolchain / package. 

Using latest version of Go, along with whatever is most recently pulled via the `go get -u and go mod tidy` process.

### PR DESCRIPTION
**Pull request by eliottness**
### What

Previously we would only check if the import was present in a fix. Now, we are checking if import is accessible as the import we are looking for.

### References

Fixes #458
Fixes #448

### CODE REVIEW

File: internal/injector/aspect/context/context.go [hunk 1]
```diff
@@ -224,8 +224,13 @@ func (c *context) ParseSource(bytes []byte) (*dst.File, error) {
...
 
-func (c *context) AddImport(path string, alias string) bool {
-	return c.refMap.AddImport(c.file, path, alias)
+func (c *context) AddImport(path string, name string) bool {
+	nodeChain := []dst.Node{c.node}
+	for p := c.NodeChain.parent; p != nil; p = p.parent {
+		nodeChain = append(nodeChain, p.node)
+	}
+
+	return c.refMap.AddImport(c.file, nodeChain, path, name)
```
Comment (RomainMuller) [inline internal/injector/aspect/context/context.go:233]: Why not just pass the `c.NodeChain`? That would be equivalent, but incur fewer allocations? (Here you're basically copying a linked list into an array list...)

File: internal/injector/testdata/injector/chi5-newroute-dotimport/config.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ aspects:
             }()
 
 syntheticReferences:
+  github.com/go-chi/chi/v5: true
```
Comment (RomainMuller) [inline internal/injector/testdata/injector/chi5-newroute-dotimport/config.yml:19]: That reference exists in the original file, so it should be be reported as a synthetic one? It's not a big deal here (it'd be filtered later one due to it already being present in the `importcfg` file, but still...)

Comment (eliottness) [reply internal/injector/testdata/injector/chi5-newroute-dotimport/config.yml:19]: This is part of the idea to mitigate shadowing when dot imports are used

File: internal/injector/aspect/context/context.go (hunk 1 above)
Comment (eliottness) [reply internal/injector/aspect/context/context.go:233]: Because of circular imports :/

**Review by RomainMuller — APPROVED** [review]

**Pull request merged**